internal/services/cdn/cdn_frontdoor_rule_resource.go

```diff
@@ -246,14 +246,14 @@ func resourceCdnFrontDoorRule() *pluginsdk.Resource {
 
 									"cdn_frontdoor_origin_group_id": {
 										Type:         pluginsdk.TypeString,
-										Required:     true,
+										Optional:     true,
 										ValidateFunc: validate.FrontDoorOriginGroupID,
 									},
 
+									// Removed Default value for issue #18889, default value is applied in the ExpandCdnFrontDoorRouteConfigurationOverrideAction func
 									"forwarding_protocol": {
 										Type:     pluginsdk.TypeString,
 										Optional: true,
-										Default:  string(cdn.ForwardingProtocolMatchRequest),
```

Comment:
This cannot be `Computed` because that would cause a perpetual diff since you would not actually be able to clear the value, which is required to make this to work correctly.

Reply:
Due to another [issue](https://github.com/hashicorp/terraform-provider-azurerm/issues/18889#issuecomment-1290253767) (mentioned in the comment thread of the original issue) I had to make every field for this `action` optional and fix it in the `expand`/`flatten` functions of the resource. This is all related to Portal magic, I have added the following to the documentation **NOTE** to the `route_configuration_override_action` block. I know this is not ideal, but I feel this is the best that we can do given the cards we have been delt.

| <b>NOTE:</b> | While `cache_duration`, `cache_behavior` and `query_string_caching_behavior` do have default values, if you do not define these properties in your configuration file you will receive a diff during plan. |
| -- | :-- |